server/ugRedisApi.js

```diff
@@ -1,107 +0,0 @@
-const log = require('@kth/log')
-const { server: serverConfig } = require('./configuration')
-
-const redisKeys = (courseCode, semester, ladokRoundIds) => ({
-  teachers: ladokRoundIds.map(round => `${courseCode}.${semester}.${round}.teachers`),
-  examiner: [`${courseCode}.examiner`],
-  responsibles: ladokRoundIds.map(round => `${courseCode}.${semester}.${round}.courseresponsible`),
-  assistants: ladokRoundIds.map(round => `${courseCode}.${semester}.${round}.assistants`), // edu.courses.SF.SF1624.20191.1.assistants
-})
-
-const _removeDublicates = personListWithDublicates =>
-  personListWithDublicates
-    .map(person => JSON.stringify(person))
-    .filter((person, index, self) => self.indexOf(person) === index)
-    .map(personStr => JSON.parse(personStr))
-
-const createPersonHtml = personList => {
-  let personString = ''
-  personList &&
-    personList.forEach(person => {
-      if (person) {
-        personString += `<p class = "person">
-      <img src="https://www.kth.se/files/thumbnail/${person.username}" alt="Profile picture" width="31" height="31">
-      <a href="/profile/${person.username}/" property="teach:teacher">
-          ${person.givenName} ${person.lastName} 
-      </a> 
-    </p>  `
-      }
-    })
-  return personString
-}
-
-// ------- EXAMINATOR AND RESPONSIBLES FROM UG-REDIS: ------- /
-async function _getCourseEmployees(apiMemoData) {
-  const { courseCode, semester, ladokRoundIds } = apiMemoData
-  try {
-    const { assistants, teachers, examiner, responsibles } = redisKeys(courseCode, semester, ladokRoundIds)
-    log.info(
-      '_getCourseEmployees for all memos course rounds with keys: ',
-      assistants,
-      teachers,
-      examiner,
-      responsibles
-    )
-    const ugClient = await redis('ugRedis', serverConfig.cache.ugRedis.redis)
-    const arrWithStringifiedArrays = await ugClient
-      .multi()
-      .mget(teachers) // [0]
-      .mget(examiner) // [1]
-      .mget(responsibles) // [2]
-      .mget(assistants) // [3]
-      .execAsync()
-    log.info(
-      ' Ug Redis fetched correctly,  number of teachers: ',
-      arrWithStringifiedArrays[0].length,
-      ' for course ',
-      courseCode,
-      ' for semester ',
-      semester
-    )
-    log.info(
-      ' Ug Redis, examiners: ',
-      arrWithStringifiedArrays[1].length,
-      ' for course ',
-      courseCode,
-      ' for semester ',
-      semester
-    )
-    log.info(
-      ' Ug Redis, responsibles: ',
-      arrWithStringifiedArrays[2].length,
-      ' for course ',
-      courseCode,
-      ' for semester ',
-      semester
-    )
-    log.info(
-      ' Ug Redis, assistants: ',
-      arrWithStringifiedArrays[3].length,
-      ' for course ',
-      courseCode,
-      ' for semester ',
-      semester
-    )
-
-    const flatArrWithHtmlStr = arrWithStringifiedArrays.map(perTypeStringifiedArr => {
-      const thisTypeAllRoundsEmployees = perTypeStringifiedArr.flatMap(perRoundStr => JSON.parse(perRoundStr))
```

Comment:
In admin pages we fetch each type of employee for each round, the user can chose to create 1 document for 3 rounds, therefore it means each type of user will come as `[  ['teacher1 round1'], ['teacher1 round2', 'teacher2 round2'], ['teacher2 round3']  ]` it means before deleting duplicates and creating html it needs to be flatted like this `[ 'teacher1 round1', 'teacher1 round2', 'teacher2 round2', 'teacher2 round3'  ]`, therefore for each array for each employee type needs to be flatted with `flatMap`

Reply:
the same for `kursutveckling-admin-web` and `kursinfo-web`

Reply:
You can try to test production rest-api it has more data, or create tests

Reply:
Added remove duplicates method to avoid copy of employee data.